app/cli/cmd/root.go

```diff
@@ -211,3 +227,39 @@ func loadControlplaneAuthToken(cmd *cobra.Command) (string, error) {
 	// loaded from config file, previously stored via "auth login"
 	return viper.GetString(confOptions.authToken.viperKey), nil
 }
+
+// ParseToken the token and return the type of token
+func ParseToken(token string) (*ParsedToken, error) {
+	if token == "" {
+		return &ParsedToken{}, nil
+	}
+
+	// Create a parser without claims validation
+	parser := jwt.NewParser(jwt.WithoutClaimsValidation())
+
+	// Parse the token without verification
+	parsedToken, _, err := parser.ParseUnverified(token, jwt.MapClaims{})
+	if err != nil {
+		return nil, err
+	}
+
+	// Extract claims
+	claims := parsedToken.Claims.(jwt.MapClaims)
+
+	// Check for specific claims and return accordingly
+	if val, ok := claims["workflow_id"]; ok && val != nil {
```

Comment:
do we have any other way to detect it? Could the audience or the issuer give you the hint? I thought the issuer actually would tell you what it is.

Reply:
The issuer is the same for all tokens.

We can potentially try the audience which values are:
- api-token-auth.chainloop
- user-auth.chainloop
- attestations.chainloop

But then the mapping of Claims make it harder but let me have a try.